libcontainer/cgroups/systemd/v1.go

```diff
@@ -222,7 +223,16 @@ func (m *legacyManager) Destroy() error {
 		return err
 	}
 	unitName := getUnitName(m.cgroups)
-	if err := stopUnit(dbusConnection, unitName); err != nil {
+
+	err = stopUnit(dbusConnection, unitName)
+	// Both on success and on error, cleanup all the cgroups we are aware of.
+	// Some of them were created directly by Apply() and are not managed by systemd.
+	for _, path := range m.paths {
+		if err := unix.Rmdir(path); err != nil && err != unix.ENOENT {
```

Comment:
:+1: for using rmdir(2)

Reply:
On a second though, we have used `cgroups.RemovePaths()` which for some historical reasons does some ugly tricks, which might be required (on some older kernels? IDK), so maybe use it instead.

Reply:
One reason to use `cgroups.RemovePaths()` is in case a container creates sub-cgroups, `rmdir` might not work (depending on the kernel I guess).

Reply:
thanks for the hint.  Fixed

Reply:
We have a test with sub-cgroups but it's v2-specific. I am writing a similar one for v1